src/schema/v2/artworkImageSearch.ts

```diff
@@ -0,0 +1,153 @@
+      resolve: ({ query_overlap_percent }) => query_overlap_percent,
+    },
+    targetOverlapPercent: {
+      description:
+        "How much of the collection image overlaps the query image, as a percentage",
+      type: new GraphQLNonNull(GraphQLFloat),
+      resolve: ({ target_overlap_percent }) => target_overlap_percent,
+    },
+    targetMatchRect: {
+      description: "Location of the matching area in the second image",
+      type: new GraphQLNonNull(ArtworkImageSearchMatchRect),
+      resolve: ({ target_match_rect }) => target_match_rect,
+    },
+    queryMatchRect: {
+      description: "Location of the matching area in the first image",
+      type: new GraphQLNonNull(ArtworkImageSearchMatchRect),
+      resolve: ({ query_match_rect }) => query_match_rect,
+    },
+  }),
+})
+
+export const ReverseImageSearchResults = new GraphQLObjectType({
+  name: "ReverseImageSearchResults",
+  fields: {
+    results: {
+      type: new GraphQLList(ReverseImageSearchResult),
+    },
+  },
+})
+
+const ArtworkImageSearchType = new GraphQLUnionType({
+  name: "ArtworkImageSearchType",
+  types: [ReverseImageSearchResults, ErrorsType],
+  resolveType: ({ __typename }) => {
+    if (__typename === "ReverseImageSearchResults") {
+      return ReverseImageSearchResults
+    }
+
+    return ErrorsType
+  },
+})
+
+export const ArtworkImageSearch: GraphQLFieldConfig<void, ResolverContext> = {
+  type: ArtworkImageSearchType,
+  description: "Search for matching artworks by image",
+  args: {
+    image: {
+      description: "Image file",
+      // TODO: Remove `as unknown` when "graphql" is updated to version 16.x.x
+      // Wrong TS declaration for GraphQLScalarType in node_modules/graphql/type/definition.d.ts
+      type: new GraphQLNonNull((GraphQLUpload as unknown) as GraphQLScalarType),
+    },
+  },
+  resolve: async (
+    _root,
+    { image },
+    { meLoader, searchByImageLoader },
+    info
+  ) => {
+    if (!meLoader) {
+      throw new Error("You need to be signed in to perform this action")
+    }
+
+    // Verifying that the token is still valid
+    try {
+      await meLoader()
+    } catch (err) {
+      throw new Error("You need to be signed in to perform this action")
+    }
+
+    const { filename, mimetype, createReadStream } = await image
+    const stream: ReadStream = createReadStream()
+
+    // Without this hack, we will get an error that we are sending an unsupported file format
+    // @ts-ignore
+    stream.path = stream?._writeStream?._path
+
+    const response = await searchByImageLoader({
+      image: stream,
+      filename,
+      contentType: mimetype,
+    })
+
+    if (response.status === "ok") {
+      return {
+        __typename: "ReverseImageSearchResults",
+        results: response.result,
+      }
+    }
+
+    return {
+      errors: response.error.map((error) => ({
+        message: error,
+        code: "invalid",
+        path: [info.fieldName],
+      })),
+    }
```

Comment:
According to the conditions of the jira ticket, we should be able to get errors in this way
```graphql
{
  artworkImageSearch(image: $file) {
    ... on Errors {
      errors {
        message
      }
    }
  }
}
```

But usually such pattern is used in resolvers in Gravity (for example, as [here](https://github.com/artsy/gravity/blob/main/app/graphql/mutations/create_user_address.rb#L13-L22)) and there are no such use cases in Metaphysics. 

**Question**: Maybe it will be enough if we just throw error in this way?

```javascript
throw new Error(response.error.join("\n"))
```